drivers/unix/file_access_unix.cpp

```diff
@@ -68,16 +68,23 @@ Error FileAccessUnix::open_internal(const String &p_path, int p_mode_flags) {
 	//printf("opening %s, %i\n", path.utf8().get_data(), Memory::get_static_mem_usage());
 
 	ERR_FAIL_COND_V_MSG(f, ERR_ALREADY_IN_USE, "File is already in use.");
-	const char *mode_string;
-
-	if (p_mode_flags == READ) {
-		mode_string = "rb";
-	} else if (p_mode_flags == WRITE) {
-		mode_string = "wb";
-	} else if (p_mode_flags == READ_WRITE) {
-		mode_string = "rb+";
-	} else if (p_mode_flags == WRITE_READ) {
-		mode_string = "wb+";
+	String mode_string = "";
+
+	if (p_mode_flags & ModeBitFields::TEMPORARY_FIELD) {
+		mode_string += "D";
```

Comment:
If `D` a standard mode option on Unix? It's not listed in macOS fdopen man page, so probably not supported at least on macOS/iOS and BSDs.

Reply:
In the code it is using fopen() not fdopen()? I don't think that should matter though. I haven't tested it on Unix yet I'll look into this. Thank you for the heads up.

Reply:
Both have the same man page and I do not see any mentions of `D` anywhere except for Windows docs.

Reply:
Thanks for the test it's hard to test this without a mac. Ill make the D mode string Windows specific. I'll look into an Apple specific method to implement. I'll keep you posted :)

Reply:
Usually, it's ok to use `unlink` on file after opening it to have the same effect.